Libraries/CLI/src/cli.c

```diff
@@ -0,0 +1,265 @@
+        beginning_space = true;
+    char *argv[10];
+    int argc = 0;
+    memset(argv, 0, sizeof(argv));
+
+    //Iterate over each character in input
+    for (p = input; p < end; p++) {
+        //If in a token
+        if (in_token) {
+            //If whitespace found, token ends
+            if (white_space_present(p)) {
+                in_token = true;
+            }
+            //Else, token continues
+            else if (white_space_not_present(p)) {
+                input = p;
+                in_token = false;
+            }
+        }
+        //If not in a token
+        else {
+            //If ENTER or whitespace found
+            if (*p == ENTER || (white_space_present(p))) {
+                //If ENTER, end of command
+                if (*p == ENTER) {
+                    *p = '\0';
+                    argv[argc++] = input;
+                    break;
+                }
+
+                //If whitespace found, end of token
+                if (white_space_present(p) && !beginning_space) {
+                    *p = '\0';
+                    argv[argc++] = input;
+                    in_token = true;
+                }
+            }
+            //If not in token and whitespace not found, new token starts
+            else if (white_space_not_present(p) && beginning_space) {
+                input = p;
+                in_token = false;
+                beginning_space = false;
+            }
+            //If not in token and whitespace not found and beginning_space is false, token continues
+            else if (white_space_not_present(p) && !beginning_space) {
+                in_token = false;
+            }
+        }
+    }
+
+    //Set last argv value to NULL
+    argv[argc] = NULL;
+
+    //If no arguments, return
+    if (argc == 0)
+        return;
+
+    bool success_flag = 0; //True if input command matches
+
+    //Iterate over all commands to check if input command matches
+    for (int i = 0; i < num_commands; i++) {
+        if (strcasecmp(argv[0], commands[i].name) == 0) {
+            //Call corresponding command's handler
+            commands[i].handler(argc, argv);
+            success_flag = 1;
+            break;
+        }
+    }
+
+    //If no commands match, print error message
+    if (success_flag == 0) {
+        printf("\n\rCommand isn't valid!\n\r");
+    }
+
+    //Print prompt
+    User_Prompt_Sequence();
+}
+
+bool white_space_present(char *p)
+{
+    return *p == SPACE || *p == TAB;
+}
+
+bool white_space_not_present(char *p)
+{
+    return *p != SPACE || *p != TAB;
+}
+
+void User_Prompt_Sequence(void)
+{
+    MXC_UART_WriteCharacter(MXC_UART_GET_UART(CONSOLE_UART), DOLLAR);
+    MXC_UART_WriteCharacter(MXC_UART_GET_UART(CONSOLE_UART), SPACE);
+}
+
+/** Clears the buffer storing each character of the user command input
+*
+* @param void
+*
+* @return void
+*/
+void Clear_buffer(void)
+{
+    for (int i = 0; i < MAX_COMMAND_LENGTH; i++) {
+        buf[i] = '\0';
+    }
+}
```

Comment:
You can use memset() function to clear the buffer instead of iterating over a for loop.